cmd/snap/cmd_model.go

```diff
@@ -125,19 +144,77 @@ func (x *cmdModel) Execute(args []string) error {
 
 	w := tabWriter()
 
+	if x.Serial && client.IsAssertionNotFoundError(serialErr) {
+		// for serial assertion, the primary keys are output (model and
+		// brand-id), but if we didn't find the serial assertion then we still
+		// output the brand-id and model from the model assertion, but also
+		// return a devNotReady error
+		fmt.Fprintf(w, "brand-id:\t%s\n", modelAssertion.HeaderString("brand-id"))
+		fmt.Fprintf(w, "model:\t%s\n", modelAssertion.HeaderString("model"))
+
+		return errors.New(devNotReadyMessage)
+	}
+
+	// the rest of this function is the main flow for outputting either the
+	// model or serial assertion in normal or verbose mode
+
 	// output the primary keys first in their canonical order
-	for _, headerName := range assertion.Type().PrimaryKey {
-		if headerName == "series" {
+	for _, headerName := range mainAssertion.Type().PrimaryKey {
+		switch headerName {
+		case "series":
 			// series can be obtained from "snap version"
 			// and given it is stuck at 16 is mainly confusing
-			continue
+		case "model":
+			if x.Serial {
+				fmt.Fprintf(w, "model:\t%s\n", mainAssertion.HeaderString("model"))
+				continue
+			}
+
+			// for the model command (not --serial) we want to conditionally
+			// use the display-name for the model if it exists (with the
+			// normal model header after the display-name)
+			modelHeader := modelAssertion.HeaderString("model")
+			if displayName := modelAssertion.HeaderString("display-name"); displayName != "" {
+				modelHeader = fmt.Sprintf("%s (%s)", displayName, modelHeader)
+			}
+			fmt.Fprintf(w, "model:\t%s\n", modelHeader)
```

Comment:
the output for "snap model (no options)" should be like "snap version", so no colons ':'

Reply:
changed